backend/inspirehep/records/api/literature.py

```diff
@@ -157,18 +157,28 @@ def update(
 
     def add_files(self, data):
         if not current_app.config.get("FEATURE_FLAG_ENABLE_FILES", False):
+            LOGGER.info("Feature flag ``FEATURE_FLAG_ENABLE_FILES`` is disabled")
             return data
 
         if "deleted" in data and data["deleted"]:
+            LOGGER.info("Record is deleted", uuid=self.id)
             return data
 
         documents = data.pop("documents", [])
         figures = data.pop("figures", [])
 
+        self.pop("documents", None)
+        self.pop("figures", None)
+        self.delete_all_files()
```

Comment:
What's the benefit of removing all files?

Reply:
my bad, that's wrong